presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -42,4 +58,111 @@ public static boolean isEsriNaN(double d)
     {
         return Double.isNaN(d) || Double.isNaN(translateFromAVNaN(d));
     }
+
+    public static int getPointCount(OGCGeometry ogcGeometry)
+    {
+        GeometryCursor cursor = ogcGeometry.getEsriGeometryCursor();
+        int points = 0;
+        while (true) {
+            com.esri.core.geometry.Geometry geometry = cursor.next();
+            if (geometry == null) {
+                return points;
+            }
+
+            if (geometry.isEmpty()) {
+                continue;
+            }
+
+            if (geometry instanceof Point) {
+                points++;
+            }
+            else {
+                points += ((MultiVertexGeometry) geometry).getPointCount();
```

Comment:
I checked type hierarchy. It is not always `MultiVertexGeometry` if it is not a `Point`. It could be `Line` for example.